triedb/pathdb/difflayer.go

```diff
@@ -232,57 +231,15 @@ func (dl *diffLayer) node(owner common.Hash, path []byte, hash common.Hash, dept
 	if ok {
 		n, ok := subset[string(path)]
 		if ok {
-			// If the trie node is not hash matched, or marked as removed,
-			// bubble up an error here. It shouldn't happen at all.
-			if n.Hash != hash {
-				dirtyFalseMeter.Mark(1)
-				log.Error("Unexpected trie node in diff layer", "owner", owner, "path", path, "expect", hash, "got", n.Hash)
-				return nil, newUnexpectedNodeError("diff", hash, n.Hash, owner, path, n.Blob)
-			}
 			dirtyHitMeter.Mark(1)
 			dirtyNodeHitDepthHist.Update(int64(depth))
 			dirtyReadMeter.Mark(int64(len(n.Blob)))
-			return n.Blob, nil
+			return n.Blob, n.Hash, &nodeLoc{loc: locDiffLayer, depth: depth}, nil
 		}
 	}
+	//TODO(will-2012): https://github.com/bnb-chain/bsc/pull/2508 broken
```

Comment:
@will-2012 metioned that
the performance by PR https://github.com/bnb-chain/bsc/pull/2508 has been roll back,
for intefaces changed

Reply:
PBSS hashcache has been re-enable by PR https://github.com/bnb-chain/bsc/pull/2765, pls review~